### PR TITLE
add feature for adding new students

### PR DESCRIPTION
**Pull request by tetianamakar**

### CODE REVIEW

File: src/main/java/com/project/attendancemanagementsystem/service/StudentService.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    public List<StudentResponse> getStudents() {
+        List<Student> students = studentRepository.findAll();
+        return students.stream()
+                .map(student -> EntityConverter.convertStudent(student))
+                .collect(Collectors.toList());
+    }
+
+    public void addStudent(StudentRequest request) {
+        Schedule existingSchedule = scheduleRepository.findById(request.getScheduleId()).orElseThrow(() -> new EntityNotFoundException("Schedule doesn't exist"));
```
Comment (vmakar1134) [inline src/main/java/com/project/attendancemanagementsystem/service/StudentService.java]: Generally, it's better not to mix repositories from different domains in a service.
If your **StudentService** has a **ScheduleRepository** then this service can potentially create or delete a schedule via **ScheduleRepository**. Therefore this line is better to move to a **ScheduleService** and use the service here.

File: src/main/java/com/project/attendancemanagementsystem/service/StudentService.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private final ScheduleRepository scheduleRepository;
+
+    public StudentService(StudentRepository studentRepository, ScheduleRepository scheduleRepository) {
+        this.studentRepository = studentRepository;
+        this.scheduleRepository = scheduleRepository;
+    }
+
+    public List<StudentResponse> getStudents() {
+        List<Student> students = studentRepository.findAll();
+        return students.stream()
```
Comment (vmakar1134) [inline src/main/java/com/project/attendancemanagementsystem/service/StudentService.java]: There is no need for a `students` variable.
You can do:
```
studentRepository.findAll().stream()
....
....
```

**Review by vmakar1134 — CHANGES_REQUESTED** [review]
Overall, very good.
I would advise some changes to the following packages:
**model** - it should contain either entities or request/response models(DTO). So you can keep requests and responses in the **model** package and create another for entities - **entity** or **domain**.
**service** - it's a bit controversial to have EntityConverter in the service layer. I would recommend moving it to its own package - **converter**.

File: src/main/java/com/project/attendancemanagementsystem/service/StudentService.java (hunk 2 above)
Comment (vmakar1134) [reply src/main/java/com/project/attendancemanagementsystem/service/StudentService.java]: Although it's not critical

**Comment by tetianamakar** [comment]
>

File: src/main/java/com/project/attendancemanagementsystem/service/StudentService.java (hunk 2 above)
Comment (tetianamakar) [reply src/main/java/com/project/attendancemanagementsystem/service/StudentService.java]: done

File: src/main/java/com/project/attendancemanagementsystem/service/StudentService.java (hunk 1 above)
Comment (tetianamakar) [reply src/main/java/com/project/attendancemanagementsystem/service/StudentService.java]: done

**Review by vmakar1134 — APPROVED** [review]

**Pull request merged**